utils.lua

```diff
@@ -50,74 +83,89 @@ function util:GetRealmSlug(realm)
+    ["pet"] = true,
+    ["vehicle"] = true,
+}
 
 do
     for i=1,40 do
-        table.insert(unitIdentifiers, "raid" .. i)
-        table.insert(unitIdentifiers, "raidpet" .. i)
-        table.insert(unitIdentifiers, "nameplate" .. i)
+        unitIdentifiers["raid" .. i] = true
+        unitIdentifiers["raidpet" .. i] = true
+        unitIdentifiers["nameplate" .. i] = true
     end
 
     for i=1,4 do
-        table.insert(unitIdentifiers, "party" .. i)
-        table.insert(unitIdentifiers, "partypet" .. i)
+        unitIdentifiers["party" .. i] = true
+        unitIdentifiers["partypet" .. i] = true
     end
 
     for i=1,5 do
-        table.insert(unitIdentifiers, "arena" .. i)
-        table.insert(unitIdentifiers, "arenapet" .. i)
+        unitIdentifiers["arena" .. i] = true
+        unitIdentifiers["arenapet" .. i] = true
     end
 
-    for i = 1, MAX_BOSS_FRAMES do
-        table.insert(unitIdentifiers, "boss" .. i)
+    for i=1, MAX_BOSS_FRAMES do
+        unitIdentifiers["boss" .. i] = true
     end
 
-    for k=1,table.getn(unitIdentifiers) do
-        table.insert(unitIdentifiers, k .. "target")
+    for k,_ in pairs(unitIdentifiers) do
+        unitIdentifiers[k .. "target"] = true
     end
 end
 
--- Can be called with isUnit("Marahin"), IsUnit("party1"))
-function util:IsUnit(identifier, isPlayer)
-    if not isPlayer and type(identifier) == "string" and identifier:find("-", nil, true) then
-        isPlayer = true
+-- Can be called with isUnit("raid14"), IsUnit("party1"))
+function util:IsUnit(identifier)
+    if identifier:find("-", nil, true) then
+        return false, false
     end
-    local isUnit = not isPlayer or type(identifier) == "string" and unitIdentifiers[identifier]
 
-    return isUnit, isUnit and UnitExists(identifier), isUnit and UnitIsPlayer(identifier)
+    -- if it exists in unitIdentifiers, it's a valid unit name (like party1 or raid20)
+    if not unitIdentifiers[identifier] then
+        return false, false
+    end
+
+    return UnitExists(identifier), UnitIsPlayer(identifier)
 end
 
-function util:GetNameRealm(arg1, arg2)
-    local unit, name, realm
-    local _, unitExists, unitIsPlayer = util:IsUnit(arg1, arg2)
+function util:GetNameRealm(characterOrUnitName, fallBackRealmName)
+    if type(characterOrUnitName) ~= "string" then
+        return
+    end
+
+    local characterName, characterRealm
+    local unitExists, unitIsPlayer = util:IsUnit(characterOrUnitName)
+    -- examples when arg1 is something like raid14 or party2
     if unitExists then
-        unit = arg1
         if unitIsPlayer then
-            name, realm = UnitName(unit)
-            realm = realm and realm ~= "" and realm or GetNormalizedRealmName()
+            characterName, characterRealm = UnitName(characterOrUnitName)
+            characterRealm = characterRealm ~= "" and characterRealm or GetNormalizedRealmName()
         end
 
-        return name, realm, unit
+        return characterName, characterRealm
```

Comment:
The unit argument was not used except for the part where it was treated as `unitExists` while it returns the first argument of the function instead.